coldfront/core/project/management/commands/deactivate_ica_projects.py

```diff
@@ -0,0 +1,182 @@
+            allocation.start_date = current_date
+            allocation.end_date = None
+            allocation.save()
+
+            message = (
+                f'Updated Project {project.name} ({project.pk})\'s status to '
+                f'{project_status.name} and Allocation {allocation.pk}\'s '
+                f'status to {allocation_status.name}.')
+
+            self.logger.info(message)
+            self.stdout.write(self.style.SUCCESS(message))
+
+    def reset_service_units(self, project, dry_run):
+        """
+        Resets service units for a project and its users to 0.00. Creates
+        the relevant transaction objects to record the change. Updates the
+        relevant historical objects with the reason for the SU change.
+
+        If dry_run is True, write to stdout without changing object fields.
+        """
+        allocation_objects = get_accounting_allocation_objects(project)
+        current_allocation = Decimal(allocation_objects.allocation_attribute.value)
+        reason = 'Resetting SUs while deactivating expired ICA project.'
+        updated_su = Decimal('0.00')
+
+        if dry_run:
+            message = f'Would reset {project.name} and its users\'s SUs from ' \
+                      f'{current_allocation} to {updated_su}. The reason ' \
+                      f'would be: "Resetting SUs while deactivating expired ' \
+                      f'ICA project."'
+
+            self.stdout.write(self.style.WARNING(message))
+
+        else:
+            set_service_units(project,
+                              allocation_objects,
+                              updated_su,
+                              reason,
+                              True)
+
+            message = f'Successfully reset SUs for {project.name} ' \
+                      f'and its users, updating {project.name}\'s SUs from ' \
+                      f'{current_allocation} to {updated_su}. The reason ' \
+                      f'was: "{reason}".'
+
+            self.logger.info(message)
+            self.stdout.write(self.style.SUCCESS(message))
+
+    def send_emails(self, project, expiry_date, dry_run):
+        """
+        Send emails to managers/PIs of the project that have notifications
+        enabled about the project deactivation.
+
+        If dry_run is True, write the emails to stdout instead.
+        """
+
+        if settings.EMAIL_ENABLED:
+            context = {
+                'project_name': project.name,
+                'expiry_date': expiry_date.strftime('%m-%d-%Y'),
+                'support_email': settings.CENTER_HELP_EMAIL,
+                'signature': settings.EMAIL_SIGNATURE,
+            }
+
+            recipients = project.managers_and_pis_emails()
+
+            if dry_run:
+                msg_plain = \
+                    render_to_string('email/expired_ica_project.txt',
+                                     context)
+
+                message = f'Would send the following email to ' \
```

Comment:
Including the email text might be overkill / not add too much value.

"Would send a notification email to k users" is fine.